pandas/core/arrays/boolean.py

```diff
@@ -108,14 +108,22 @@ def __from_arrow__(
         """
         import pyarrow
 
-        if array.type != pyarrow.bool_():
+        if array.type != pyarrow.bool_() and not pyarrow.types.is_null(array.type):
             raise TypeError(f"Expected array of boolean type, got {array.type} instead")
 
         if isinstance(array, pyarrow.Array):
             chunks = [array]
+            length = len(array)
         else:
             # pyarrow.ChunkedArray
             chunks = array.chunks
+            length = array.length
+
+        if pyarrow.types.is_null(array.type):
```

Comment:
I'd move this up before the isinstance checks

Reply:
I think the isinstance checks are necessary since the way we find length depends on the type of pyarrow array(regular array uses len, while chunkedarray needs length).

I could be missing something again, though.

Reply:
I assumed that array would have length as well, but that's not the case. So looks good